### PR TITLE
Refactor git option parser and fix wrong behaviours

### PR DESCRIPTION
**Pull request by chitoku-k**
The current option parser looks like it has reached its limitation because it did some tricky regular expressions to handle short and long options. I have entirely rewritten the core parser to properly handle the cases described in #40 and #41 and I added the tests that cover them.

### CODE REVIEW

File: src/git.zsh [hunk 1]
```diff
@@ -88,47 +86,92 @@ _fzf_complete_git() {
...
+
+            -m|--message)
+                _fzf_complete_git-commit-messages '' $@
+                ;;
+
+            --author|--date)
+                ;;
+
+            -F|-t|--file|--pathspec-from-file|--template)
+                _fzf_path_completion "${prefix/--*=}" $@$prefix_option
```
Comment (hexium310) [inline src/git.zsh]: `-Ffoo**` isn't behave as expected if the second argument of `_fzf_path_completion` is `${prefix/--*=}`. Can you change it as below?
https://github.com/chitoku-k/fzf-zsh-completions/blob/9529d750e763ef8e32501ae27faeb9183aeacb4a/src/git.zsh#L113

Comment (hexium310) [reply src/git.zsh]: `$num` is the position where short option that takes a value appears.

Comment (chitoku-k) [reply src/git.zsh]: I don't want `_fzf_complete_git_option_prefix` to revive as I removed it. I think I need to add some support anyway.

Comment (hexium310) [reply src/git.zsh]: Sorry. It is not second but first.

Comment (chitoku-k) [reply src/git.zsh]: I fixed it. Can you check whether it works?

Comment (hexium310) [reply src/git.zsh]: Looks good 💪

**Pull request merged**